### PR TITLE
[FLINK-12359][metrics][tests] Harden SystemResourcesMetricsITCase

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

Hardens the `SystemResourcesMetricsITCase` against
* slow starts of a TaskManager, in which case metrics could be registered later than the test expects
* the host portion of the metric identifier not being `localhost`

We now exclude the host from the identifier, and generate a future in the reporter that we wait on instead.

### CODE REVIEW

**Comment by rmetzger** [comment]
@flinkbot approve description consensus

File: flink-tests/src/test/java/org/apache/flink/runtime/metrics/SystemResourcesMetricsITCase.java [hunk 1]
```diff
@@ -136,8 +126,17 @@ public void close() {
 			OPENED_REPORTERS.remove(this);
 		}
 
-		public Map<Gauge<?>, String> getGauges() {
-			return gauges;
+		@Override
+		public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
+			for (final String expectedPattern : patternFutures.keySet()) {
+				if (group.getMetricIdentifier(metricName, name -> name).matches(expectedPattern)) {
```
Comment (tillrohrmann) [inline flink-tests/src/test/java/org/apache/flink/runtime/metrics/SystemResourcesMetricsITCase.java]: I think we could move `group.getMetricIdentifier()` outside of the loop

File: flink-tests/src/test/java/org/apache/flink/runtime/metrics/SystemResourcesMetricsITCase.java [hunk 2]
```diff
@@ -102,8 +94,8 @@ public void startTaskManagerAndCheckForRegisteredSystemMetrics() throws Exceptio
 		};
 
 		String[] expectedHosts = new String[] {
```
Comment (tillrohrmann) [inline flink-tests/src/test/java/org/apache/flink/runtime/metrics/SystemResourcesMetricsITCase.java]: Could be replaced with array initializer.

Comment (tillrohrmann) [reply flink-tests/src/test/java/org/apache/flink/runtime/metrics/SystemResourcesMetricsITCase.java]: The same applies to the `expectedGauges`.

**Pull request merged**